opentelemetry-sdk/tests/metrics/test_metrics.py

```diff
@@ -38,6 +38,28 @@ def test_resource_empty(self):
         # pylint: disable=protected-access
         self.assertIs(meter.resource, resources._EMPTY_RESOURCE)
 
+    def test_start_pipeline(self):
+        exporter = mock.Mock()
+        meter_provider = metrics.MeterProvider()
+        meter = meter_provider.get_meter(__name__)
+        # pylint: disable=protected-access
+        meter_provider.start_pipeline(meter, exporter, 6)
+        self.assertEqual(len(meter_provider._exporters), 1)
+        self.assertEqual(len(meter_provider._controllers), 1)
+        meter_provider.shutdown()
```

Comment:
this should be wrapped in try: finally block, so that meter provider will shut down even if an assertion fails:

```
meter_provider.start_pipeline(...)
try:
    # asserts
finally:
    meter_provider.shutdown()
```